build.rs

```diff
@@ -19,30 +19,91 @@
 use std::env;
 
+// Copy-pasted from `https://github.com/paritytech/parity-ethereum/blob/cb03f380ab2bb37ff18771e6886c42098ad8b15a/docker/android/Dockerfile#L44-L62`
+const ANDROID_CFG: [&str; 5] = [
+	"git checkout 83d918449f22720d84a341a05e24b6d109e6d3ae",
+	"git apply ../libudev.patch",
+	"./autogen.sh",
+	"./configure --disable-introspection --disable-programs --disable-hwdb --host=arm-linux-androideabi --prefix=/opt/ndk-standalone/sysroot/usr/ --enable-shared=false",
+	"make"
+];
+
+const LINUX_CFG: [&str; 3] = [
+	"./autogen.sh",
+	"./configure",
+	"make"
+];
+
 fn main() {
-    let target = env::var("TARGET").unwrap();
-
-    if target.contains("windows") {
-        cc::Build::new()
-            .file("etc/hidapi/windows/hid.c")
-            .include("etc/hidapi/hidapi")
-            .compile("libhidapi.a");
-        println!("cargo:rustc-link-lib=setupapi");
-
-    } else if target.contains("darwin") {
-        cc::Build::new()
-            .file("etc/hidapi/mac/hid.c")
-            .include("etc/hidapi/hidapi")
-            .compile("libhidapi.a");
-        println!("cargo:rustc-link-lib=framework=IOKit");
-        println!("cargo:rustc-link-lib=framework=CoreFoundation");
-
-    } else if target.contains("linux") {
-        let mut config = cc::Build::new();
-        config.file("etc/hidapi/linux/hid.c").include("etc/hidapi/hidapi");
-        config.compile("libhidapi.a");
-        println!("cargo:rustc-link-lib=udev");
-    }
+	let target = env::var("TARGET").unwrap();
+
+	if target.contains("windows") {
+		cc::Build::new()
+			.file("etc/hidapi/windows/hid.c")
+			.include("etc/hidapi/hidapi")
+			.compile("libhidapi.a");
+		println!("cargo:rustc-link-lib=setupapi");
+
+	} else if target.contains("darwin") {
+		cc::Build::new()
+		.file("etc/hidapi/mac/hid.c")
+		.include("etc/hidapi/hidapi")
+		.compile("libhidapi.a");
+
+		println!("cargo:rustc-link-lib=framework=IOKit");
+		println!("cargo:rustc-link-lib=framework=CoreFoundation");
+
+	} else if target.contains("android") {
+		set_android_flags();
+		execute_shell_cmd(&ANDROID_CFG);
+		let mut config = cc::Build::new();
+		config.file("etc/hidapi/linux/hid.c").include("etc/hidapi/hidapi");
+		config.compile("libhidapi.a");
+
+		println!("cargo:rustc-link-search=native=./etc/eudev/src/libudev/.libs");
+		println!("cargo:rustc-link-lib=static=udev");
+	} else if target.contains("linux") {
+		execute_shell_cmd(&LINUX_CFG);
+		let mut config = cc::Build::new();
+		config.file("etc/hidapi/linux/hid.c").include("etc/hidapi/hidapi");
+		config.compile("libhidapi.a");
+
+		println!("cargo:rustc-link-search=native=./etc/eudev/src/libudev/.libs");
+		println!("cargo:rustc-link-lib=static=udev");
+	}
+}
+
+fn set_android_flags() {
+	env::set_var("CFLAGS", "-D LINE_MAX=2048 -D RLIMIT_NLIMITS=15 -D IPTOS_LOWCOST=2 -std=gnu99");
+	env::set_var("CXX", "arm-linux-androideabi-clang++");
+	env::set_var("CC", "arm-linux-androideabi-gcc");
+}
+
+fn execute_shell_cmd(commands: &[&str]) {
```

Comment:
I don't have time to look in details, but that's not so great because we won't be able to compile from Windows.

Reply:
Yeah, maybe it was poorly described PR but it was intended to Linux only, I haven't looked into `Windows or OSX`

However, I found the autoconf crate which could probably simplify this mess a bit